mono/metadata/boehm-gc.c

```diff
@@ -34,6 +34,11 @@
 #include <mono/utils/mono-os-mutex.h>
 #include <mono/utils/mono-counters.h>
 
+#ifdef _MSC_VER
+	// Quiet Visual Studio linker warning, LNK4221, in cases when this source file intentional ends up empty.
+	void __mono_win32_boehm_gc_quiet_lnk4221(void) {}
```

Comment:
You can use `#pragma warning(disable: 4221)` to disable the warning, which IMHO looks clearer than the fake declaration.